### DISCUSSION TITLE
Any plans on providing Debian/Ubuntu packages?

### DISCUSSION THREAD

**Discussion by thediveo** [bpftrace/bpftrace#3378] [Q&A]
Do you plan to provide Debian/Ubuntu packages?

I've tried the AppImage version and while it works well on one Ubuntu 24.04 installation, it doesn't work on another Ubuntu 24.04 installation. The problem might be related to an AppImage integrator, but from the error message it is unclear if it is and manually starting the bpftrace AppImage always results in a cryptic error message that doesn't seem to come from the AppImage itself.

**Comment by ajor** [comment]
I'm not sure I understand - what's wrong with this Ubuntu package? https://packages.ubuntu.com/noble/bpftrace

But it's normally up to each Linux distribution to package bpftrace. We don't do that ourselves here.

**Reply by thediveo** [reply]
Its version 😁 ... I need the `for` loop construct introduced in v0.21.0 at this time, and it's unclear if and when the noble package will catch up.

**Reply by ajor** [reply]
You'd have to poke the Ubuntu or Debian package maintainers if you want them to update then.

Can you build bpftrace from source instead? That might be easier

**Reply by thediveo** [reply]
Yes, I built from sources, albeit the Dockerfile instructions did not work on a plain Ubuntu host system. There were some issues with outdated clang versions. In general, I would like to send the colleagues at work to an official deb download link that after installation simply works and is recent. Debian "LTS" is "long-term stuck" in the past and Ubuntu won't really update until the next 6 month cycle. So having official and very recent distro packages would really help, as colleagues then simply install updated packages and can make use of `bpftrace`.

**Reply by ajor** [reply]
There was an upstream Ubuntu packaging bug we hit last week that is probably what meant the Dockerfile didn't work for you. We worked around it in #3365 and our Ubuntu Dockerfile is building successfully in CI again: https://github.com/bpftrace/bpftrace/actions/runs/10229272996/job/28302731084

**Comment by ajor** [comment]
What was the issue you had with the AppImage? Getting that working might be the best solution for you since we only publish AppImage releases and don't plan to make per-distro packages ourselves

**Reply by thediveo** [reply]
The problem was actually not with the bpftrace AppImage itself, but with https://github.com/TheAssassin/AppImageLauncher; I've removed it and now I can properly run this CLI appimage.
